CHANGELOG.md

```diff
@@ -5,6 +5,59 @@ All notable changes to **RTVI Client Web** will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [0.2.0] - 2024-09-13
+
+RTVI 0.2.0 removes client-side configuration to ensure state management is handled exclusively by the bot. Clients no longer maintain an internal config array that can be altered outside of a ready state. Developers needing stateful configuration before a session starts should implement it independently.
+
+This change enforces a key design principle of the RTVI standard: the bot should always be the single source of truth for configuration, and clients should remain stateless.
+
+This release also preludes the implementation of a `TextClient` by renaming classes and types away from `VoiceClientX` to `RTVIClientX`. Where possible, we have left aliased deprecations to support backwards compatibility. 
+
+### Added
+
+- `params` client constructor param, a partial object that will be sent as JSON stringified body params at `start()` to your hosted endpoint. If you want to declare initial configuration in your client, or specify start services on the client, you can declare them here.
+- `onConfig` and `VoiceEvents.Config` callback & event added, triggered by `getConfig` voice message.
+- `@transportReady` decorator added to methods that should only be called at runtime. Note: decorator support required several Parcel configuration changes and additional dev dependencies.
+- `@getIfTransportInState` getter decorator added to getter methods that should only be called in a specified transport state.
+
+### Changed
+
+- `start()` has been renamed to `connect()`.
+- Client no longer expects a `services` map as a constructor param (note: remains in place but flagged as deprecated.) If you want to pass a services map to your endpoint, please use `params`.
+- `customHeaders` has been renamed to `headers`.
+- Config getter and setter methods (`getConfig` and `updateConfig`) are only supported at runtime. 
+- `updateConfig` promise is typed to `Promise<VoiceMessage>` (previously `unknown` to support offline updates.)
+- `getConfig` promise is typed to `Promise<VoiceClientOptions[]>` (previously `unknown` to support offline updates.)
+- `services` getter and setter methods have been deprecated.
+- `getServiceOptionsFromConfig`, `getServiceOptionValueFromConfig`, `setConfigOptions` and `setServiceOptionInConfig` are now async to support `getConfig` at runtime and accept an optional `config` param for working with local config arrays.
```

Comment:
Better to just always require the `config` parameter, and never fetch the config inside the method? (and therefore make it non-async?) Otherwise I can see people calling this multiple times to fetch several values, and causing a round trip each time.

It would be nice to separate these from the `RTVIClient` entirely and have them be utility functions in a separate namespace.